File: Easy/Math/1979. Find Greatest Common Divisor of Array.py
# 1979. Find Greatest Common Divisor of Array
# app->1
class Solution:
    import math
    def findGCD(self, nums: List[int]) -> int:
        nums.sort()
        return math.gcd(nums[0], nums[-1])


# app->2
class Solution:
    import math
    def findGCD(self, nums: List[int]) -> int:
        big = max(nums)
        small = min(nums)
        return math.gcd(big,small)

        

File: Easy/Math/1979. Find Greatest Common Divisor of Array.cpp
//1979. Find Greatest Common Divisor of Array 
class Solution {
public:
    int findGCD(vector<int>& nums) {
        int small = nums[0];
        int big = nums[0];
        int n = nums.size();
        for(int i = 1; i<n;i++){
            small = min(small,nums[i]);
            big = max(big,nums[i]);

        }
        return __gcd(small,big);
        
    }
};
